end-to-end-tests/code-build/buildspec.yml

```diff
@@ -0,0 +1,26 @@
+version: 0.2
+
+phases:
+  pre_build:
+    commands:
+      - mkdir -p /root/.gradle
+      - echo "org.gradle.daemon=false" > /root/.gradle/gradle.properties
+      - cat end-to-end-tests/code-build/end2end.properties >> /root/.gradle/gradle.properties
+      - aws configure --profile default set region us-west-2
```

Comment:
Are we fixed with the region "us-west-2"? Do we allow customized regions

Reply:
This one most likely will not to be reused to setup multiple regions.  It's mostly for rebuild/recover/update the test environment. So far it's only for **us-west-2**.

Actually this command line is tricky one `aws configure --profile default set region us-west-2`, it will force to create a 'default` profile.  Because if you specify `--profile default`, it requires a default profile exists even it will try to follow the credential chain to find credential.

Reply:
Understood. 
I think the `aws configure --profile default set region us-west-2` will not create the default profile if it already exists? And if the user have already created the stacks, he/she should already have the default profile :-D

Reply:
yep, it will not create a new one if it exists.  This configuration file is used by CodeBuild, so every time is a completely new environment.  In this context, it will actually create a new one.

The reason we want to use `default` profile is we need to use the default credential chain, which will look for the instance profile (or technically it's container credential profile). See https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/credentials.html#credentials-default